zebrad/tests/acceptance.rs

```diff
@@ -351,3 +351,61 @@ fn valid_generated_config(command: &str, expected_output: &str) -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn listener_address_test() -> Result<()> {
+    listener_address(
+        "localhost:8233",
+        Some("Opened Zcash protocol endpoint at 127.0.0.1:8233"),
+    )?;
+    listener_address(
+        "127.0.0.1:8233",
+        Some("Opened Zcash protocol endpoint at 127.0.0.1:8233"),
+    )?;
+    listener_address(
+        "::1:8233",
+        Some("Opened Zcash protocol endpoint at \\[::1\\]:8233"),
+    )?;
+    listener_address("invalid", None)?;
+
+    Ok(())
+}
+
+fn listener_address(request: &str, expected_response: Option<&str>) -> Result<()> {
+    zebra_test::init();
+
+    // Create a temp dir to work in
+    let dir = TempDir::new("zebrad_tests")?;
+
+    // Create a config file with request as listen_addr
+    let mut config = ZebradConfig::default();
+    config.state.cache_dir = dir.path().join("state");
+    config.state.memory_cache_bytes = 256000000;
+    config.network.listen_addr = request.parse()?;
+    let config_path = dir.path().join("zebrad.toml");
+    fs::File::create(&config_path)?.write_all(toml::to_string(&config)?.as_bytes())?;
+
+    // Start zebrad with created config and kill it at 10 seconds.
```

Comment:
10 seconds is a long time.

Resolving localhost should be instant, and the IP addresses just use string parsing.
Do you know what is taking so long?

(The invalid case will block the thread, and so it might not fail before the child is killed. We should accept a hang or a panic in that test.)

Reply:
I am not totally sure what is going on here so still researching. On my pc 1 sec is not enough to get zcash protocol output so 3 seconds work. In the CI i tried even 20 seconds and it is hanging there, at: 
- https://github.com/ZcashFoundation/zebra/pull/951/checks?check_run_id=1032440268#step:6:1345
```
   Aug 26 16:18:30.357  INFO zebrad::commands::start: Starting zebrad
   Aug 26 16:18:30.357  INFO zebrad::commands::start: starting to connect to the network self=StartCmd { filters: [] }
   Aug 26 16:18:30.538  INFO zebra_consensus::chain: initialising ChainVerifier network=Mainnet max_checkpoint_height=Some(Height(420347)) initial_height=None initial_hash=None
```

This is just before what i see locally:

```
Aug 26 13:04:12.051  INFO zebra_consensus::chain: initialising ChainVerifier network=Mainnet max_checkpoint_height=Some(Height(420347)) initial_height=Some(Height(56000)) initial_hash=Some(BlockHeaderHash("d85373ba68f0cb6d5d718c89537241eb8d2b40add956b02e0beace4e00000000"))
Aug 26 13:04:13.858  INFO zebra_network::peer_set::initialize: Sending initial request for peers
Aug 26 13:04:13.859  INFO listen{addr=V4(127.0.0.1:4441)}: zebra_network::peer_set::initialize: Opened Zcash protocol endpoint at 127.0.0.1:4441
```

Reply:
I think the IPv6 address in the config is not formatted correctly, it needs square brackets:
https://github.com/ZcashFoundation/zebra/pull/951/files#r477657791

Reply:
I changed that, thanks. However there is another issue here as all tests fail in the CI:

https://github.com/ZcashFoundation/zebra/pull/951/checks?check_run_id=1036323722#step:6:1190

Reply:
I'm not sure, I can't see anything obvious in the code.

Can you try running zebrad locally, with logging set to trace?
What is the last log line that it outputs?

Can yo try logging some output before the address is resolved, and before the peer set is opened?
That might help you work out which step is slow.

Reply:
It seems the problem is not being caused by this PR specifically but with a connectivity issue in the CI. Looks like it cant do anything that requires internet. Locally all this just works.

I made a small test at https://github.com/oxarbitrage/zebra/pull/4 trying to get a bigger output. The output is now truncated at https://github.com/oxarbitrage/zebra/pull/4/checks?check_run_id=1036756131#step:6:1216
We can open the port because no resolve is involved, however that is the last thing it can do as later activity requires the internet.

Reply:
It's strange that the listen address always uses the network.
The IP addresses should parse, without using the network at all.
I'll go read the underlying source code to see what's happening.

Can we just test that the output logs the port being opened?

Reply:
Yeah, it tries to parse IP addresses first:
https://doc.rust-lang.org/src/std/net/addr.rs.html#975

So I'm really not sure why we're seeing these errors with the IP addresses.

Reply:
It seems like a bad sign that we now have to have a network connection in order to parse our configs.  I wonder whether this is a sign that we should pause and rethink this feature.